File: 프로그래머스/3/12938. 최고의 집합/최고의 집합.js
function solution(n, s) {
    if (n > s) return [-1];
    
    const [div, mod] = [Math.floor(s / n), s % n];
    const answer = Array(n).fill(div);
    
    for (let i = 0; i < mod; i++) {
        answer[n - 1 - i] += 1;
    }
    return answer;
}

File: 프로그래머스/3/12938. 최고의 집합/최고의 집합.py
def solution(n, s):
    if n > s:
        return [-1]
    
    q, r = divmod(s, n)
    answer = [q] * n
    
    # 나머지(r)만큼의 개수만큼 뒤에서부터 1씩 더해주면 자연스럽게 오름차순이 유지됨
    for i in range(r):
        answer[n - 1 - i] += 1
        
    return answer